C++ API parity: CTCLoss

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #28783 C++ API parity: BCEWithLogitsLoss
* #28918 C++ API parity: isfinite
* #28755 C++ API parity: PoissonNLLLoss
* **#28654 C++ API parity: CTCLoss**

Differential Revision: [D18202437](https://our.internmc.facebook.com/intern/diff/D18202437)

pbelevich committed:
C++ API parity: CTCLoss

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"




[ghstack-poisoned]

yf225 commented on test/cpp/api/functional.cpp:
```diff
@@ -1382,3 +1382,69 @@ TEST_F(FunctionalTest, Pad) {
+      const auto targets = torch::randint(1, 15, {9}, torch::kLong);
+      const auto log_probs =
+        torch::randn({50, 3, 15}, torch::kDouble).log_softmax(2);
+      const auto loss = torch::nn::functional::ctc_loss(
```
It would be awesome to change all `torch::nn::functional::ctc_loss` references in this file to `F::ctc_loss` :D

yf225 reviewed (approved):
@pbelevich Thanks so much for the awesome work! I left a minor comment.

yf225 commented:
Also it would be awesome to update the parity tracker `test/cpp_api_parity/parity-tracker.md` for this module as well :D

pbelevich committed:
Update on "C++ API parity: CTCLoss"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"


Differential Revision: [D18202437](https://our.internmc.facebook.com/intern/diff/D18202437)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"


Differential Revision: [D18202437](https://our.internmc.facebook.com/intern/diff/D18202437)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"


Differential Revision: [D18202437](https://our.internmc.facebook.com/intern/diff/D18202437)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: CTCLoss"


Differential Revision: [D18202437](https://our.internmc.facebook.com/intern/diff/D18202437)

[ghstack-poisoned]
